Teardown Failing with "IndexError: list index out of range" after Setup Failure

in RDR testruns, the teardown phase of a test is failing with below error when there is a prior error during the setup phase. This behavior prevents proper cleanup of resources, leading to issues in subsequent test runs.

```
    def teardown():
        failed_to_delete = False
        for instance in instances:
            try:
>               instance.delete_workload(switch_ctx=ctx[0], force=True)
E               IndexError: list index out of range

tests/conftest.py:6535: IndexError
```

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -6487,6 +6487,7 @@ def dr_workload(request):
             list: objects of workload class.
 
         """
+        ctx.append(switch_ctx)
         total_pvc_count = 0
         workload_key = "dr_workload_subscription"
         if pvc_interface == constants.CEPHFILESYSTEM:
@@ -6522,7 +6523,6 @@ def dr_workload(request):
                 dr_helpers.wait_for_mirroring_status_ok(
                     replaying_images=total_pvc_count
                 )
-        ctx.append(switch_ctx)
         return instances
 
     def teardown():

